src/components/platform-admin/controllers.tsx

```diff
@@ -103,6 +177,168 @@ export async function createOrganization(
+    logger: ctx.app.logger,
+  });
+
+  const orgsList = (await cf.v3Organizations()).filter(
+    org => !org.name.match(/^(CATS|ACC|BACC|SMOKE|PERF|AIVENBACC|ASATS)-/),
+  );
+
+  const template = new Template(ctx.viewContext, TITLE_EMAIL_ORG_MANAGERS);
+
+return {
+    body: template.render(<ContactOrganisationManagersPage
+      linkTo={ctx.linkTo}
+      csrf={ctx.viewContext.csrf}
+      orgs={orgsList}
+    />),
+  };
+}
+
+// when form is submitted
+export async function contactOrganisationManagersPost(
+  ctx: IContext, _params: IParameters, body: IContactOrganisationManagersPageValues,
+): Promise<IResponse> {
+  throwErrorIfNotAdmin(ctx);
+
+  const cf = new CloudFoundryClient({
+    accessToken: ctx.token.accessToken,
+    apiEndpoint: ctx.app.cloudFoundryAPI,
+    logger: ctx.app.logger,
+  });
+
+  const uaa = new UAAClient({
+    apiEndpoint: ctx.app.uaaAPI,
+    clientCredentials: {
+      clientID: ctx.app.oauthClientID,
+      clientSecret: ctx.app.oauthClientSecret,
+    },
+  });
+
+  const accountsClient = new AccountsClient({
+    apiEndpoint: ctx.app.accountsAPI,
+    logger: ctx.app.logger,
+    secret: ctx.app.accountsSecret,
+  });
+
+  const errors = [];
+
+  // check if any fields are missing
+  // if so, create error messages
+  errors.push(
+    ...validateOrgSelection(body),
+    ...validateManagerRoleSelection(body),
+    ...validateEmailMessage(body),
+  );
+
+  const template = new Template(ctx.viewContext);
+  const orgsList = (await cf.v3Organizations()).filter(
+    org => !org.name.match(/^(CATS|ACC|BACC|SMOKE|PERF|AIVENBACC|ASATS)-/),
+  );
+  let orgUserEmails:ReadonlyArray<any> = [];
+
+  // get email adfresses for all managers for the selected type and organisation
+  if (body.organisation) {
+    const usersForGivenOrg = await cf.usersForOrganization(body.organisation);
+    // we need to look into paas-accounts for their actual email address
+    const filteredManagers = await Promise.all(
+      usersForGivenOrg.filter(user => user.entity.organization_roles.includes(body.managerRole))
+        .map(async manager => await accountsClient.getUser(manager.metadata.guid)),
+    );
+
+    // create a zendesk api-friendly array of email address objects
+    orgUserEmails = Array.from(
+      new Set(
+        filteredManagers
+          .filter((user): user is IAccountsUser => !!user)
+          .map(user => ({ user_email: user.email })),
+      ),
+    );
+
+    // if manager role type has been selected but there are no users with selected role, update the error message
+    // message property is readonly so cannot just update the message
+    if (orgUserEmails.length === 0 && body.managerRole ) {
+      /* istanbul ignore next */
+      const managerRoleErrorIndex = errors.findIndex(
+        error => error.field === 'managerRole',
+      );
+      errors.splice(managerRoleErrorIndex, 0, { field: 'managerRole', message: 'Select organisation does not have any selected manager roles' });
```

Comment:
I find this message a bit confusing. To me it sounds like I haven't picked a role, rather than "there is no one with this role"
```suggestion
      errors.splice(managerRoleErrorIndex, 0, { field: 'managerRole', message: 'Select organisation does not have any of the selected manager roles' });
```